apps/server/src/modules/board/service/drawing-do-authorizable.service.spec.ts

```diff
@@ -0,0 +1,209 @@
+				const firstNames = boardDoAuthorizable.users.reduce((map, user) => {
+					map[user.userId] = user.firstName;
+					return map;
+				}, {});
+
+				const lastNames = boardDoAuthorizable.users.reduce((map, user) => {
+					map[user.userId] = user.lastName;
+					return map;
+				}, {});
+
+				expect(boardDoAuthorizable.users).toHaveLength(5);
+				expect(firstNames[teacher.id]).toEqual(teacher.firstName);
+				expect(lastNames[teacher.id]).toEqual(teacher.lastName);
+				expect(firstNames[substitutionTeacher.id]).toEqual(substitutionTeacher.firstName);
+				expect(lastNames[substitutionTeacher.id]).toEqual(substitutionTeacher.lastName);
+				expect(firstNames[students[0].id]).toEqual(students[0].firstName);
+				expect(lastNames[students[0].id]).toEqual(students[0].lastName);
+				expect(firstNames[students[1].id]).toEqual(students[1].firstName);
+				expect(lastNames[students[1].id]).toEqual(students[1].lastName);
+				expect(firstNames[students[2].id]).toEqual(students[2].firstName);
+				expect(lastNames[students[2].id]).toEqual(students[2].lastName);
+			});
+
+			it('should return the boardDo', async () => {
+				const { board } = setup();
+
+				const boardDoAuthorizable = await service.getBoardAuthorizable(board);
+
+				expect(boardDoAuthorizable.boardDo).toEqual(board);
+			});
+
+			it('should return the parentDo', async () => {
+				setup();
+				const column = columnFactory.build();
+				const card = cardFactory.build();
+
+				drawingDoRepo.findParentOfId.mockResolvedValueOnce(column);
+
+				const boardDoAuthorizable = await service.getBoardAuthorizable(card);
+
+				expect(boardDoAuthorizable.parentDo).toEqual(column);
+			});
+		});
+
+		describe('when trying to create a boardDoAuthorizable on a column without a columnboard as root', () => {
+			const setup = () => {
+				const roles = roleFactory.buildList(1, {});
+				const teacher = userFactory.buildWithId({ roles });
+				const students = userFactory.buildListWithId(3);
+				const column = columnFactory.build();
+				drawingDoRepo.findById.mockResolvedValueOnce(column);
+				return { column, teacherId: teacher.id, studentIds: students.map((s) => s.id) };
+			};
+
+			it('should throw an error', async () => {
+				const { column } = setup();
+
+				await expect(() => service.getBoardAuthorizable(column)).rejects.toThrowError();
+			});
+		});
+
+		describe('when having a board that does not reference a course', () => {
+			const setup = () => {
+				const teacher = userFactory.buildWithId();
+				const board = columnBoardFactory.withoutContext().build();
+				drawingDoRepo.findById.mockResolvedValueOnce(board);
+				return { board, teacherId: teacher.id };
+			};
+
+			it('should return an boardDoAuthorizable without user-entries', async () => {
+				const { board } = setup();
+
+				const boardDoAuthorizable = await service.getBoardAuthorizable(board);
+				expect(boardDoAuthorizable.users).toHaveLength(0);
+			});
+		});
+	});
+});
```

Comment:
I think we are missing a clear api tests for all this. 
An api test should make it clear that the student cannot edit the drawing element, but can edit files belonging to it